### DISCUSSION TITLE
Problem contacting remote_ts lan with established ike v2 tunnel with fortigate

### DISCUSSION THREAD

**Discussion by gregou2007** [strongswan/strongswan#1307] [General]
Hello,

I have managed to set up a ikeV2 tunnel between an openwrt router (linksys) and a fortigate firewall.
Both have Public IP adresses. the set up is site to site vpn (tunnel) with behind each local subnets which need to communicate.

**Informations on set up**  

lan (192.168.1.X) -----------(Openwrt router 192.168.1.1 // wan public adress lets call it X.X.X.X)------------- (wan remote lets call it Y.Y.Y.Y //Fortigate 192.168.2.254) --------(lan 192.168.2.X)

I have searched a lot and i didnt managed to find a way  to ping from 192.168.1.1 the fortigate 192.168.2.254 or even a 192.168.2.X adress (all rules are ok on fortigate)

but I can ping 192.168.1.1 from a 192.168.2.X address, i can't ping any other 192.168.1.X address

When the tunnel is set up a strange think is that a route is added on table 220 which seem erroneous (it says to go through the wan gateway of openwrt wan)

root@OpenWrt:~# ip route show table 220
192.168.2.0/24 via WAN GW IP dev wan proto static src 192.168.1.1

What do i have to do in order to access 192.18.2.X lan from 192.168.1.X lan and vice versa ?

additional informations :

root@OpenWrt:~# ipsec statusall
Status of IKE charon daemon (strongSwan 5.9.5, Linux 5.10.138, armv7l):
  uptime: 6 minutes, since Sep 17 05:03:42 2022
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 2
  loaded plugins: charon aes des rc2 sha2 sha1 md5 random nonce x509 revocation constraints pubkey pkcs1 pgp dnskey sshkey pem fips-prf gmp gmpdh xcbc hmac attr kernel-netlink resolve socket-default connmark stroke vici updown xauth-generic
Listening IP addresses:
  Y.Y.Y.Y
  192.168.1.1
  fd52:98df:49f3::1
Connections:
         net:  %any...Y.Y.Y.Y  IKEv2
         net:   local:  uses pre-shared key authentication
         net:   remote: [Y.Y.Y.Y] uses pre-shared key authentication
       net-1:   child:  192.168.1.0/24 === 192.168.2.0/24 TUNNEL
Routed Connections:
       net-1{1}:  ROUTED, TUNNEL, reqid 1
       net-1{1}:   192.168.1.0/24 === 192.168.2.0/24
Security Associations (1 up, 0 connecting):
         net[1]: ESTABLISHED 6 minutes ago, X.X.X.X[X.X.X.X]...Y.Y.Y.Y[Y.Y.Y.Y]
         net[1]: IKEv2 SPIs: 43dac47514896c44_i 63dcf7cf52c64ab6_r*, rekeying in 3 hours
         net[1]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
       net-1{2}:  INSTALLED, TUNNEL, reqid 1, ESP SPIs: c5649cfd_i 100182cb_o
       net-1{2}:  AES_CBC_256/HMAC_SHA2_256_128, 0 bytes_i, 0 bytes_o, rekeying in 50 minutes
       net-1{2}:   192.168.1.0/24 === 192.168.2.0/24

root@OpenWrt:~# ip route
default via 128.106.X.Y dev wan proto static src 128.106.X.Y (wan)
128.106.X.0/24 dev wan proto kernel scope link src 128.106.X.48
192.168.1.0/24 dev br-lan proto kernel scope link src 192.168.1.1

-I'm sure fortigate has a good configuration

stwanctl.conf

# Section defining IKE connection configurations.

connections {
    net {
       remote_addrs  = Y.Y.Y.Y
       local {
          auth = psk
       }
       remote {
          auth = psk
          id = Y.Y.Y.Y
       }
       children {
          net-1 {
             local_ts  = 192.168.1.0/24
             remote_ts  = 192.168.2.0/24
             start_action = trap
             dpd_action = restart
             esp_proposals = aes256-sha256-modp2048
          }
       }
       version = 2
       proposals = aes256-sha256-modp2048
    }
 }
 secrets {
    ike-net {
       id = XXXXXXXXXXX
       secret = XXXXXXXXXXXXX
    }
 }

thanks

**Comment by Thermi** [comment]
Two ways:
1) probably needs a NAT exception: pastebin full firewall rules of all tables, probably you need to except anything with an IPsec policy in *nat POSTROUTING from the nat
2) use route based IPsec so you have a different interface, then you can also have a second firewall zone in UCI. To implement that you would use an XFRM interface, that should be usable on your kernel (did not verify kernel version).

**Reply by gregou2007** [reply]
Hello thank you
1. here what i have in the firewall : 


root@OpenWrt:~# cat /etc/config/firewall

config defaults
        option input 'ACCEPT'
        option output 'ACCEPT'
        option forward 'REJECT'
        option synflood_protect '1'

config zone
        option name 'lan'
        option input 'ACCEPT'
        option output 'ACCEPT'
        option forward 'ACCEPT'
        list network 'lan'

config zone
        option name 'wan'
        option input 'REJECT'
        option output 'ACCEPT'
        option forward 'REJECT'
        option masq '1'
        option mtu_fix '1'
        list network 'wan'
        list network 'wan6'

config forwarding
        option src 'lan'
        option dest 'wan'

config rule
        option name 'Allow-DHCP-Renew'
        option src 'wan'
        option proto 'udp'
        option dest_port '68'
        option target 'ACCEPT'
        option family 'ipv4'

config rule
        option name 'Allow-Ping'
        option src 'wan'
        option proto 'icmp'
        option icmp_type 'echo-request'
        option family 'ipv4'
        option target 'ACCEPT'

config rule
        option name 'Allow-IGMP'
        option src 'wan'
        option proto 'igmp'
        option family 'ipv4'
        option target 'ACCEPT'

config rule
        option name 'Allow-DHCPv6'
        option src 'wan'
        option proto 'udp'
        option dest_port '546'
        option family 'ipv6'
        option target 'ACCEPT'

config rule
        option name 'Allow-MLD'
        option src 'wan'
        option proto 'icmp'
        option src_ip 'fe80::/10'
        list icmp_type '130/0'
        list icmp_type '131/0'
        list icmp_type '132/0'
        list icmp_type '143/0'
        option family 'ipv6'
        option target 'ACCEPT'

config rule
        option name 'Allow-ICMPv6-Input'
        option src 'wan'
        option proto 'icmp'
        list icmp_type 'echo-request'
        list icmp_type 'echo-reply'
        list icmp_type 'destination-unreachable'
        list icmp_type 'packet-too-big'
        list icmp_type 'time-exceeded'
        list icmp_type 'bad-header'
        list icmp_type 'unknown-header-type'
        list icmp_type 'router-solicitation'
        list icmp_type 'neighbour-solicitation'
        list icmp_type 'router-advertisement'
        list icmp_type 'neighbour-advertisement'
        option limit '1000/sec'
        option family 'ipv6'
        option target 'ACCEPT'

config rule
        option name 'Allow-ICMPv6-Forward'
        option src 'wan'
        option dest '*'
        option proto 'icmp'
        list icmp_type 'echo-request'
        list icmp_type 'echo-reply'
        list icmp_type 'destination-unreachable'
        list icmp_type 'packet-too-big'
        list icmp_type 'time-exceeded'
        list icmp_type 'bad-header'
        list icmp_type 'unknown-header-type'
        option limit '1000/sec'
        option family 'ipv6'
        option target 'ACCEPT'

config rule
        option name 'Allow-IPSec-ESP'
        option src 'wan'
        option dest 'lan'
        option proto 'esp'
        option target 'ACCEPT'

config rule
        option name 'Allow-ISAKMP'
        option src 'wan'
        option dest 'lan'
        option dest_port '500'
        option proto 'udp'
        option target 'ACCEPT'

config rule 'ipsec_esp'
        option src 'wan'
        option name 'IPSec ESP'
        option proto 'esp'
        option target 'ACCEPT'

config rule 'ipsec_ike'
        option src 'wan'
        option name 'IPSec IKE'
        option proto 'udp'
        option dest_port '500'
        option target 'ACCEPT'

config rule 'ipsec_nat_traversal'
        option src 'wan'
        option name 'IPSec NAT-T'
        option proto 'udp'
        option dest_port '4500'
        option target 'ACCEPT'

config rule 'ipsec_auth_header'
        option src 'wan'
        option name 'Auth Header'
        option proto 'ah'
        option target 'ACCEPT'





It seems not to use iptables  :   

root@OpenWrt:~# iptables -L -v -n
Chain INPUT (policy ACCEPT 419K packets, 139M bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain FORWARD (policy ACCEPT 12M packets, 12G bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 447K packets, 62M bytes)
 pkts bytes target     prot opt in     out     source               destination
root@OpenWrt:~# iptables -S
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT


3. My kernel and swanctl version :

Linux OpenWrt 5.10.138 
swanctl 5.9.6

**Reply by Thermi** [reply]
Thank you for the output, it's not telling me anything useful though. Can you run `nft list ruleset` (or whatever the right syntax is to show all rules for nftables).

Alternatively just start configuring an XFRM interface. That also takes care of potential MSS/MTU issues really quickly. For your use case (site-to-site tunnel), it's the better alternative.

**Comment by gregou2007** [comment]
Hello it took time but i managed via xfrm interfaces. thanks a lot. Though i dont manage via /etc/config/network to create the equivalent of that if you now ?

commands i have to set 
ip link add ipsec0 type xfrm dev lo if_id 1
ip link set ipsec0 up
ip route add 192.168.2.0/24 dev ipsec0 src 192.168.1.1

my /etc/config/network which doesnt work  : 

config 'interface' 'ipsec0'
  option 'ifid' '1'
  option 'mtu' '1438'
  option 'zone' 'vpn'
  option 'proto' 'xfrm'
  # useful if you want to run Bonjour/mDNS across VPN tunnels
  option 'multicast' 'true'
 
config 'route'
  option 'interface' 'ipsec0'
  option 'target' '192.168.2.0/24'
  option 'source' '192.168.1.0/24'

**Comment by Thermi** [comment]
Congratulations! You probably want to lower the MTU of the xfrm interface to like 1400 or 1300 to take the overhead into account.

I can not help you with the UCI setup of the interface exactly, maybe you need to upgrade it too. I did not check when support for it was introduced in it.
